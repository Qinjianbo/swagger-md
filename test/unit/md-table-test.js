import createTable from '../../src/generators/md-table';

describe('test/unit/md-table-test.js', () => {

  describe('createTable', () => {

    describe('with simple headers and records', () => {

      it('should return a markdown table', () => {
        expect(createTable(['a', 'b', 'c'], [
          { a: 1, b: 2, c: 3 },
          { a: 4, b: 5, c: 6 },
        ])).toMatchSnapshot();
      });

    });

    describe('when one of the headers is never used', () => {

      it('should not include the column', () => {
        expect(createTable(['a', 'b', 'c'], [
          { a: 1, b: undefined, c: 3 },
          { a: 4, b: undefined, c: 6 },
        ])).toMatchSnapshot();
      });

    });

    describe('alignment', () => {

      it('should align to a wide header', () => {
        expect(createTable(['a', 'b', 'wide_header'], [
          { a: 1, b: 2, wide_header: 3 },
          { a: 4, b: 5, wide_header: 6 },
        ])).toMatchSnapshot();
      });

      it('should align to a wide record value', () => {
        expect(createTable(['a', 'b', 'wide_header'], [
          { a: 'lots of text', b: 2, wide_header: 3 },
          { a: 4, b: 'another long string', wide_header: 6 },
        ])).toMatchSnapshot();
      });

    });

  });

});
