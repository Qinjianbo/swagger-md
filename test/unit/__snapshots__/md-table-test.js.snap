exports[`test/unit/md-table-test.js createTable alignment should align to a wide header 1`] = `
"| a | b | wide_header |
|---|---|-------------|
| 1 | 2 | 3           |
| 4 | 5 | 6           |"
`;

exports[`test/unit/md-table-test.js createTable alignment should align to a wide record value 1`] = `
"| a            | b                   | wide_header |
|--------------|---------------------|-------------|
| lots of text | 2                   | 3           |
| 4            | another long string | 6           |"
`;

exports[`test/unit/md-table-test.js createTable when one of the headers is never used should not include the column 1`] = `
"| a | c |
|---|---|
| 1 | 3 |
| 4 | 6 |"
`;

exports[`test/unit/md-table-test.js createTable with simple headers and records should return a markdown table 1`] = `
"| a | b | c |
|---|---|---|
| 1 | 2 | 3 |
| 4 | 5 | 6 |"
`;
